pkg/hintrunner/zero/zerohint_memcpy.go

```diff
@@ -3,9 +3,58 @@ package zero
 import (
 	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/core"
 	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	"github.com/NethermindEth/cairo-vm-go/pkg/utils"
+	"github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+
 	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
 )
 
+func newMemcpyContinueCopyingHinter(n, dst hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "MemcpyContinueCopying",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//n -= 1
+			//ids.continue_copying = 1 if n > 0 else 0
+			var lhs fp.Element
+
+			// n-=1
+			n, err := ctx.ScopeManager.GetVariableValue("n")
+			if err != nil {
+				return err
+			}
+			ctx.ScopeManager.AssignVariable("n", lhs.Sub(n.(*fp.Element), &utils.FeltOne))
+
+			// ids.continue_copying = 1 if n > 0 else 0
+			continue_copying, err := dst.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			var v memory.MemoryValue
+			if !utils.FeltIsPositive(n.(*fp.Element)) {
```

Comment:
You want to check here the condition for `n>0`, see that this is different from checking `FeltIsPositive`.
`FeltIsPositive` looks like the following:
```
func FeltIsPositive(felt *fp.Element) bool {
	// range_check_builtin.bound is utils.FeltMax128 (1 << 128).
	return FeltLt(felt, &FeltMax128)
}
```
This means it checks the number is smaller than 1<<128, which works on felts to determine whether you'd consider the number positive or negative. However, even if you initialized `n` with a value that fits in a felt, in the python code of the hint, `n` is treated as an integer, not a felt, so you are actually comparing it to 0.

So the correct condition here for `n>0` should be something like `FeltLt(FeltZero, n)`

Reply:
ty for the explanation! I corrected the if statement! ✅